ebu_tt_live/ui/user_input_producer/uip.js

```diff
@@ -0,0 +1,570 @@
+                     sequence["frame_rate"] = $("#frame-rate-input").val();
+                     sequence["frm_numerator"] = $("#frm-numerator-input").val();
+                     sequence["frm_denominator"] = $("#frm-denominator-input").val();
+                     sequence["marker_mode"] = $("#marker-mode-selector").val();
+                     sequence["drop_mode"] = $("#drop-mode-selector").val();
+                  }
+                  all_sequences[sequence_identifier] = sequence;
+                  sequence_numbers[sequence_identifier] = 1;
+                  if (typeof(Storage) !== "undefined") {
+                     localStorage.sequence_selector = $("#sequence-selector").html();
+                     localStorage.all_sequences = JSON.stringify(all_sequences);
+                     localStorage.sequence_numbers = JSON.stringify(sequence_numbers);
+                  }
+                  hideNewSequenceDiv();
+                  notifySuccess($("#new-sequence-notification-span"), "New sequence created", true);
+               }
+            });
+
+            // Reset the whole page
+            $("#reset-all").click(function(){
+               $("#sequence-selector").html("");
+               if (typeof(Storage) !== "undefined") {
+                  $("#subtitle-form-notification-div").text("");
+                  localStorage.clear();
+                  sequence_numbers = {};
+                  all_sequences = {};
+               }
+            });
+
+            // Encoding function from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent
+            function fixedEncodeURIComponent (str) {
+              return encodeURIComponent(str).replace(/[!'()*]/g, function(c) {
+                return '%' + c.charCodeAt(0).toString(16);
+              });
+            }
+
+            // websocket connection
+            $("#websocket-connect-button").click(function(){
+               if (socket.connected == false) {
+                  var websocket_url = $("#websocket-url-input").val() + "/" + fixedEncodeURIComponent($('#sequence-selector').val()) + "/publish";
+                  //console.log("WebSocket URL: ", websocket_url);
+                  socket.websocket = new WebSocket(websocket_url);
+                  // onclose hook is not set yet because we need to know if the connection has been established
+                  // and to get an error if not. However if onclose is defined it is called instead of onerror.
+                  // So we define onclose in the onopen hook.
+                  socket.websocket.onerror = websocketOnError;
+                  socket.websocket.onopen = websocketOnOpen;
+                  socket.websocket.onmessage = websocketOnMessage;
+               }
+            });
+
+            // websocket disconnection
+            $("#websocket-disconnect-button").click(function(){
+               if (socket.connected == true) {
```

Comment:
inconsistent? `if (socket.connected)` would match the rest of the document style